corehq/apps/reports_core/filters.py

```diff
@@ -277,16 +277,19 @@ class ChoiceListFilter(BaseFilter):
     """
     template = 'reports_core/filters/choice_list_filter.html'
 
-    def __init__(self, name, datatype, label='Choice List Filter',
+    def __init__(self, name, field, datatype, label='Choice List Filter',
                  css_id=None, choices=None):
+        from corehq.apps.userreports.reports.filters.choice_providers import StaticChoiceProvider
         params = [
             FilterParam(name, True),
         ]
         super(ChoiceListFilter, self).__init__(name=name, params=params)
+        self.field = field
         self.datatype = datatype
         self.label = label
         self.css_id = css_id or self.name
         self.choices = choices or []
+        self.choice_provider = StaticChoiceProvider(self.choices)
```

Comment:
I see, that makes sense. In some ways it'd be nice if instead of a different class, you had a function `get_choice_list_filter` that returns a `DynamicChoiceListFilter` that has a `StaticChoiceProvider` `choice_provider`. Or something. But it looks like there are a few more differences between them in code, and while a more explicit approach would be be to use an abstract base class to define the interface, I'm not sure it makes sense and whether that would actually make things clearer or not. Anyway, thanks for the explanation!